.github/workflows/check-pr.yaml

```diff
@@ -55,3 +55,244 @@ jobs:
+                      else:
+                          status = "⚡ Slight change"
+                      
+                      comment += f"| `{result['name']}` | **{change_pct:+.2f}%** | [{lower_pct:+.2f}%, {upper_pct:+.2f}%] | {status} |\n"
+                  
+                  comment += "\n"
+              
+              if has_regression:
+                  comment += "⚠️ **Warning:** Performance regressions detected (>5% slower with 95% confidence)!\n\n"
+              
+              comment += "<details>\n<summary>How to interpret these results</summary>\n\n"
+              comment += "- **Change**: Estimated performance difference (negative = faster, positive = slower)\n"
+              comment += "- **Confidence Interval**: 95% confidence bounds for the change\n"
+              comment += "- **Regression**: Lower bound > +5% (confidently slower)\n"
+              comment += "- **Improvement**: Upper bound < -5% (confidently faster)\n\n"
+              comment += "Criterion uses statistical analysis to account for noise and provide reliable comparisons.\n"
+              comment += "</details>\n"
+
+          # Write outputs
+          with open(os.environ['GITHUB_OUTPUT'], 'a') as f:
+              f.write(f"comment<<EOF\n{comment}\nEOF\n")
+              f.write(f"has_regression={'true' if has_regression else 'false'}\n")
+          EOF
+
+      - name: Comment PR with results
+        uses: actions/github-script@v7
+        env:
+          COMMENT_BODY: ${{ steps.report.outputs.comment }}
+        with:
+          script: |
+            const comment = process.env.COMMENT_BODY;
+
+            // Find existing benchmark comment
+            const { data: comments } = await github.rest.issues.listComments({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number,
+            });
+
+            const botComment = comments.find(comment => 
+              comment.user.type === 'Bot' && 
+              comment.body.includes('📊 Benchmark Results')
+            );
+
+            if (botComment) {
+              // Update existing comment
+              await github.rest.issues.updateComment({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                comment_id: botComment.id,
+                body: comment
+              });
+            } else {
+              // Create new comment
+              await github.rest.issues.createComment({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: context.issue.number,
+                body: comment
+              });
+            }
+
+      - name: Upload benchmark reports
+        uses: actions/upload-artifact@v4
+        with:
+          name: criterion-reports
+          path: |
+            dotlottie-rs/target/criterion/
+          retention-days: 30
+
+      - name: Check for regressions
```

Comment:
[Suggestion]
I don’t think we need this step. Automatically blocking PRs on any regression isn’t ideal, some performance regressions can be acceptable.